spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/HttpClientProperties.java

```diff
@@ -293,6 +296,61 @@ public String toString() {
 		/** The default ssl configuration type. Defaults to TCP. */
 		private SslProvider.DefaultConfigurationType defaultConfigurationType = SslProvider.DefaultConfigurationType.TCP;
 
+		/** Keystore path for Netty HttpClient. */
+		private String keyStore;
+
+		/** Keystore type for Netty HttpClient, default is JKS. */
+		private String keyStoreType = "JKS";
+
+		/** Keystore provider for Netty HttpClient, optional field. */
+		private String keyStoreProvider;
+
+		/** Keystore password. */
+		private String keyStorePassword;
+
+		/** Key password, default is same as keyStorePassword. */
+		private String keyPassword;
```

Comment:
There are duplicate password fields. It is never used other than to try it first, then the keyStorePassword. Show me where it is otherwise used.

Reply:
PKCS12 format keystore restricts from having a different key password, but in JKS you can have a different key password, consider the new commit.

Reply:
@spencergibb 
any update or review comment on this one?